ts/test/mock_transpiler.bzl

```diff
@@ -21,15 +23,38 @@ def _mock_impl(ctx):
             inputs = [src],
             outputs = [js_file],
             command = "cp $@",
-            arguments = [src.path, js_file.path.replace(".ts", ".js")],
+            arguments = [src.path, js_file.path],
         )
 
         ctx.actions.write(
             output = map_file,
             content = _DUMMY_SOURCEMAP % src.short_path,
         )
 
-    return DefaultInfo(files = depset(out_files))
```

Comment:
Why did this have to switch? The transpiler should just have to output the transpiled js or dts (+maps) as default info?

Reply:
Having `JsInfo` seems typical of actual usage. Some examples:
- https://github.com/aspect-build/rules_swc/blob/9498a51454efc51ec5c2da9bac9fad610a3f4eab/swc/private/swc.bzl#L362-L375
- https://github.com/aspect-build/rules_esbuild/blob/49510f4eaaab95a5c637dbdacc9f7bc9c80406fd/esbuild/private/esbuild.bzl#L446-L458

Reply:
That just means the rule can be used directly for things like `js_library(deps)` I guess, instead of how `ts_project` (the macro) or `js_library(srcs)` uses the `DefaultInfo`.

I guess it does not harm, but also isn't used in these tests 🤷

Reply:
Yeah it's probably worth doing in case folks crib this as an example (though we could also provide a production-ready transpiler using oxc as an example ;)